vendor/k8s.io/kubernetes/pkg/registry/generic/registry/store.go

```diff
@@ -872,7 +872,7 @@ func (e *Store) WatchPredicate(ctx api.Context, m *generic.SelectionPredicate, r
 
 	if name, ok := m.MatchesSingle(); ok {
 		if key, err := e.KeyFunc(ctx, name); err == nil {
-			if err != nil {
+			if err != nil { // TODO Dead code?
```

Comment:
Good catch, it looks like this piece of code should be:
```
	if name, ok := p.MatchesSingle(); ok {
		key, err := e.KeyFunc(ctx, name)
		if err != nil {
			return nil, err
		}
		w, err := e.Storage.Watch(ctx, key, resourceVersion, p)
		if err != nil {
			return nil, err
		}
		if e.Decorator != nil {
			return newDecoratedWatcher(w, e.Decorator), nil
		}
		return w, nil
		// if we cannot extract a key based on the current context, the optimization is skipped
	}
```
but I'll defer to @smarterclayton who introduced that change.